Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,51 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+
+test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
 
+test("given an empty array, returns 0", () => {
+    expect(sum([])).toBe(0);
+});
+
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
 
+test("given an array containing negative numbers, returns the correct total sum", () => {
+    expect(sum([-10, -20, 30])).toBe(0);
+    expect(sum([-1, -2, -3])).toBe(-6);
+});
+
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
 
+test("given an array with decimal/float numbers, returns the correct total sum", () => {
+    expect(sum([1.5, 2.5, 3.5])).toBe(7.5);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Note: Values with decimal part equals to `.5` happened to be able to be represented exactly in floating point number format.

Reply:
Thank you @cjyuan, i did some research in the matter and have found that using .toBeCloseTo() would fix the problem of the test failing

    expect(sum([1.1, 1.1, 1.1])).toBeCloseTo(3.3, 5);

The 5 is the tolerance for decimal places. In this case it would mean that if the result is off by 0.00001, the test will still pass.